src/super_context.ml

```diff
@@ -343,7 +380,52 @@ let create
       make ~config:context
         ~inherit_from:(Some (lazy (make ~inherit_from:None
                                      ~config:workspace)))
-  ) in
+  )
+  in
+  let artifacts_decl = Fdecl.create () in
```

Comment:
I wonder if we couldn't simplify this code by restricting what variables are allowed inside the `dst` fields of install stanzas with `(section bin)` even more. In practice, the only variable the user will want to use are things like `ext_obj`.

Reply:
I pushed a version of this.

It clearly needs some clean up (this `do_expand_var` stuff seems messy and it has more "assert false" than necessary). Otherwise, is this roughly the right direction?

Reply:
Yh, also I feel like it's a case where we can simplify things by cutting through the layers. I gave it a shot

Reply:
Oh, this does indeed look much simpler. Thank you!